app/src/ui/create-branch/create-branch-dialog.tsx

```diff
@@ -392,13 +401,27 @@ export class CreateBranch extends React.Component<
     )
 
     // there are no relevant rules for this branch name, so return
-    if (toCheckForBypass.size === 0) {
+    if (toCheck.size === 0) {
+      return
+    }
+
+    // check for actual failures
+    const { branchNamePatterns, creationRestricted } = await parseRepoRules(
```

Comment:
This is conducted one other place in `app-store` for cacheing to reduce number of times it is called for the `commit-messages` section - I believe this is mostly an general efficiency measure asa it doesn't cause more heavy resource things like api call. I am safe guarding against unnecessary running of this by the above check that says to only run this if there are rules that apply to branch names.

https://github.com/desktop/desktop/blob/b794fa55beffffb5b4c242ff830f8283a7ee92c9/app/src/lib/stores/app-store.ts#L1193-L1206

The below `getFailedRules` is called on every `commit-messages.tsx` render so I am not as concerned about it getting called more frequently, and should be the call of one regex anyways.

https://github.com/desktop/desktop/blob/b794fa55beffffb5b4c242ff830f8283a7ee92c9/app/src/ui/changes/commit-message.tsx#L445-L467

Reply:
This is perfectly fine to call more often like you said, I just cached it because it doesn't _need_ to run more often in most cases. The only thing that can change it is the underlying rulesets changing, and that can only happen when they're queried in `app-store.ts` in the first place.